### PR TITLE
expression: implement vectorized evaluation for `builtinRandomBytesSig`

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve? 
implement vectorized evaluation for builtinRandomBytesSig, for #12106
### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinEncryptionFunc -run=BenchmarkVectorizedBuiltinEncryptionFunc -args 'builtinRandomBytesSig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinEncryptionFunc/builtinRandomBytesSig-VecBuiltinFunc-4         	     355	   3391971 ns/op	   34870 B/op	    1027 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinRandomBytesSig-NonVecBuiltinFunc-4      	     318	   3750427 ns/op	 1113363 B/op	    2048 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	3.138s
```
### Check List 
Tests 
- Unit test

### CODE REVIEW

**Comment by mmyj** [comment]
The unit test case always failed because this expression make a new random []byte every time.
What should I do to fix it in this case?

File: expression/builtin_encryption_vec_test.go [hunk 1]
```diff
@@ -22,13 +22,15 @@ import (
...
-	ast.RandomBytes:        {},
+	ast.AesEncrypt: {},
+	ast.Uncompress: {},
+	ast.AesDecrypt: {},
+	ast.Compress:   {},
+	ast.MD5:        {},
+	ast.SHA:        {},
+	ast.RandomBytes: {
+		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETInt}, geners: []dataGenerator{&rangeInt64Gener{1, 1024}}},
+	},
```
Comment (qw4990) [inline expression/builtin_encryption_vec_test.go]: The reason causing unit test problems is that our testing framework compares results of vectorized functions and row-based functions to check if there is any bug.
For builtin functions like `Rand`, `RandBytes`, we can't use this way to test them.

So I think you can remove these lines.

**Comment by qw4990** [comment]
@mmyj Friendly ping, please address comments and resovle conflicts.

File: expression/builtin_encryption_vec_test.go (hunk 1 above)
Comment (mmyj) [reply expression/builtin_encryption_vec_test.go]: ok, I will fix it, and resolve conflicts.

**Comment by mmyj** [comment]
@qw4990 PTAL

File: expression/builtin_encryption_vec.go [hunk 2]
```diff
@@ -131,11 +134,40 @@ func (b *builtinAesDecryptIVSig) vecEvalString(input *chunk.Chunk, result *chunk
...
 
 func (b *builtinRandomBytesSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRandomBytesSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
```
Comment (Reminiscent) [inline expression/builtin_encryption_vec.go]: Here should be ``` buf, err := b.bufAllocator.get(types.ETInt, n)```

Comment (mmyj) [reply expression/builtin_encryption_vec.go]: I will fix it.

**Comment by qw4990** [comment]
@mmyj sorry for my late reply, this PR seems good to me, but please solve conflicts~

**Pull request merged**